### PR TITLE
Redirect taxonomic IDs using NCBI Taxonomy DB

### PR DESCRIPTION
**Pull request by priyappillai**
Using the [NCBI Taxonomy Database](https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi), find and use the most recent taxonomic ID to look up sequences. 

While taxonomic ID are currently being redirected via the viral genome database, there are cases where the viral genome database's redirects are not fully up to date; the taxonomy database's redirects are more consistent.

### CODE REVIEW

File: adapt/prepare/ncbi_neighbors.py [hunk 1]
```diff
@@ -138,6 +138,34 @@ def read_taxid_from_ncbi_url(url):
...
+    """Determine NCBI's current taxonomic ID given an (old) taxonomic ID
+
+    Args:
+        given_taxid: previously used NCBI taxonomic ID
+
+    Returns:
+        most current NCBI taxonomic ID
+    """
+    taxid = given_taxid
+    taxid_in_xml = None
```
Comment (haydenm) [inline adapt/prepare/ncbi_neighbors.py:151]: I don't think you need to set this?

File: adapt/prepare/ncbi_neighbors.py [hunk 2]
```diff
@@ -138,6 +138,34 @@ def read_taxid_from_ncbi_url(url):
...
+        if taxid_in_xml != 0:
+            logger.warning("Taxid %d is being redirected to taxid %d" %
+                           (taxid, taxid_in_xml))
+            taxid = taxid_in_xml
+        else:
+            logger.debug("Taxid %d did not need to be redirected" %
+                       (taxid))
+        redirects += 1
+    return taxid
+
```
Comment (haydenm) [inline adapt/prepare/ncbi_neighbors.py:168]: Super minor, but we've generally been doing two new lines between functions.

**Review by haydenm — APPROVED** [review]
Looks good :+1:  I think just make sure ADAPT fails gracefully when the provided taxid doesn't exist.

File: adapt/prepare/ncbi_neighbors.py (hunk 1 above)
Comment (priyappillai) [reply adapt/prepare/ncbi_neighbors.py:151]: `taxid_in_xml` should be `aka_taxid_in_xml`; it won't be set to the correct tax ID if there is no redirecting. Will rename! I could change the `given_taxid` reference if that's what you mean, but I'd rather not modify an input reference

**Pull request merged**